packages/synapse-react-client/src/utils/hooks/useDetectSSOCode.ts

```diff
@@ -87,105 +89,108 @@ export default function useDetectSSOCode(
     return null
   }, [isHandlingSynapseOAuthSignIn, searchParams])
 
-  const { accessToken } = useSynapseContext()
   const [isLoading, setIsLoading] = useState(!!(code && provider))
 
   useEffect(() => {
-    if (code && provider) {
-      const redirectUrl = `${redirectURL}?provider=${provider}`
+    if (hasInitializedSession) {
+      if (code && provider) {
+        const redirectUrl = `${redirectURL}?provider=${provider}`
 
-      //If user is already logged in, and the provider is ORCID, then try to bind this OAuth provider to the account.
-      if (OAUTH2_PROVIDERS.ORCID == provider && accessToken !== undefined) {
-        // now bind this to the user account
-        const onFailure = (err: SynapseClientError) => {
-          console.error('Error binding ORCiD to account: ', err)
-          if (onError) {
-            onError(err.reason)
-          }
-        }
-        bindOAuthProviderToAccount(
-          provider,
-          code,
-          redirectUrl,
-          BackendDestinationEnum.REPO_ENDPOINT,
-        )
-          .then(onSignInComplete)
-          .catch(onFailure)
-          .finally(() => setIsLoading(false))
-      } else if (
-        OAUTH2_PROVIDERS.GOOGLE == provider ||
-        OAUTH2_PROVIDERS.ORCID == provider
-      ) {
-        const onSuccess = (
-          response: LoginResponse | TwoFactorAuthErrorResponse | null,
-        ) => {
-          if (response) {
-            if ('accessToken' in response) {
-              setAccessTokenCookie(response.accessToken).then(onSignInComplete)
-            } else {
-              // The app will redirect or open a modal to handle a standard 2FA sign in
-              if (onTwoFactorAuthRequired) {
-                onTwoFactorAuthRequired(response)
-              }
-              if (
-                // The user logged in with OAuth while attempting to disable 2FA using an emailed signed token
-                state &&
-                state.twoFaResetToken &&
-                onTwoFactorAuthResetTokenPresent
-              ) {
-                // Let the app handle redirecting to the 2FA reset page
-                onTwoFactorAuthResetTokenPresent(
-                  response,
-                  state.twoFaResetToken,
-                )
-              }
+        //If user is already logged in, and the provider is ORCID, then try to bind this OAuth provider to the account.
+        if (OAUTH2_PROVIDERS.ORCID == provider && token !== undefined) {
```

Comment:
The assumption here is that the access token has been initialized by this time (when this is mounted), which may not be true!  So I updated this hook to only execute after the session has been initialized so we can determine what action to take if the provider is ORCID.  If an access token is present, then try to bind to the account.  If not present, then try to log in (utilizing the previously bound alias).